src/spectaql/index.js

```diff
@@ -49,13 +49,15 @@ function run(opts) {
   const { introspectionResponse, graphQLSchema } = buildSchemas(opts)
 
   // Figure out what data arranger to use...the default one, or the one from the theme
-  const customDataArrangerExists = ['data/index.js', 'data.js'].some(
+  const customDataArrangerSuffixThatExists = ['data/index.js', 'data.js'].find(
     (pathSuffix) => {
       return fileExists(path.normalize(`${themeDir}/${pathSuffix}`))
     }
   )
-  const arrangeDataModule = customDataArrangerExists
-    ? require(path.normalize(`${themeDir}/data`))
+  const arrangeDataModule = customDataArrangerSuffixThatExists
+    ? await import(
+        path.normalize(`${themeDir}/${customDataArrangerSuffixThatExists}`)
+      )
```

Comment:
Wow that's it eh? Now it can be imported in both CJS and ESM projects?

Reply:
I believe the accurate statement is this:

In SpectaQL 2x which is babelized for CJS and requires a Node version >= 14, this change allows the usage of Themes that are written in ESM.


In general, the dynamic import pattern here can be used anywhere to import ESM modules in a CJS module running on 14+, but most of those imports occur at the top level of a module so I think it would require a non trivial change. As you see here I has to make the run function async